concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -1614,7 +1614,12 @@ newAddValidator pbs ai va@ValidatorAdd{..} = do
                   _poolCommissionRates = vaCommissionRates
                 }
     let bakerInfo = bakerKeyUpdateToInfo bid vaKeys
-    let bakerInfoEx = BaseAccounts.BakerInfoExV1 bakerInfo poolInfo
+    let bakerInfoEx =
+            BaseAccounts.BakerInfoExV1
+                { _bieBakerPoolInfo = poolInfo,
+                  _bieBakerInfo = bakerInfo,
+                  _bieAccountIsSuspended = conditionally hasValidatorSuspension False
```

Comment:
As an argument for `newAddValidator`? Adding them suspended by default would be quiet surprising.

Reply:
When you register a validator, if you specify the suspended state then it should be respected. Someone might register their validator as suspended initially for instance if their node won't have caught up by the time it would be in the committee, or if they won't have had time to put their keys in their node. However, I think for backwards compatibility we shouldn't require the suspended status to be set when adding a validator, and default to not suspended in that case.

Reply:
I believe that Thomas has a good argument here although I was also in doubt during my review.

Reply:
I also agree that we should respect the suspended state given at registration. Currently, this is not supported however. I think it would be best to implement this as a separate feature outside of this PR.

Reply:
Fine with me to introduce it in another PR :blush:

Reply:
Tracked in #1246 and commented in code.